File: techs/gcd.cc
#include<iostream>
using namespace std;

typedef long long ll;

ll gcd(ll A, ll B) {
    return B? gcd(B, A%B) : A;
}


int main() {
    cout << gcd(18, 12) << endl;
    cout << gcd(12, 18) << endl;

    return 0;
}

File: techs/gcd.py
def gcd(m, n):
    if n == 0:
        return m

    return gcd(n, m % n)


print(gcd(15, 6))
print(gcd(51, 15))
print(gcd(15, 51))
print(gcd(1071, 1029))
